### PR TITLE
ESLint, Prettier, and CI Installation / Configuration

### LINKED ISSUE #2: Configure ESLint + Prettier
# Goal

Install ESLint and Prettier on frontend repository to ensure consistent styling & catch code quality issues / potential bugs.

### PR DESCRIPTION
**Pull request by mpolisch**
## Summary
This PR installs & configures prettier, adds it to package.json, and adds a simple ci.yml script to run eslint, prettier, and typechecking when opening a PR and pushing to main.

Also updated readme to list run commands.

Closes #2 

## Approach & Design Decisions
Made various formatting decisions for prettier:
- Semicolons required at the end of statements
- Single quotes standard
- Trailing commas valid for objects & arrays
- max warnings of 0 for eslint (removes possibility of linting debt)

## Security Considerations
None really. CI pipeline increases reliability.

## Testing
**What's covered:**
- [ ] Unit tests
- [ ] Edge cases (list below)

**Edge cases considered:**
None, no unit tests either (nothing to cover).

## Checklist
- [x] No secrets or credentials in code
- [] Input validation in place
- [ ] Error handling covers failure paths
- [x] Code is self-documenting or commented where non-obvious
- [ ] No dead code or debug logs left in

## Notes
Nothing to note

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+              run: npm ci
+
+            - name: Run ESLint
+              run: npm run lint
+
+            - name: Check Prettier formatting
+              run: npm run format:check
+
+            - name: Type check
+              run: npm run typecheck
```
[the remark this answers is not in the document]
Comment (mpolisch) [reply .github/workflows/ci.yml]: added typecheck rule in package.json
